tutorials/profiler/model.py

```diff
@@ -0,0 +1,257 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
This seems like gpt-fast right? In which case we have a fork already in https://github.com/pytorch/ao/tree/main/torchao/_models/llama

Reply:
It's a simplified version with lots of features stripped out and cleaner printing to demonstrate usage of the performance counter.  

See the `Usage` section of the PR and the [README](https://github.com/pytorch/ao/blob/9e2790d2a996267cf5907284e7c9f51fca53c09a/tutorials/profiler/README.md) for more details.

Reply:
I see! I will be out this week for a friend's wedding but @andrewor14 mind reviewing this PR?

Reply:
@msaroufim @andrewor14 

Let me know if this is along the lines of what you had in mind regarding #426.  

The core abstractions `DeviceSpec` and `TransformerPerformanceCounter` take care of tracking the necessary measurements (achieved BW and FLOPs/s) and detecting device-specific BW and FLOPs/s for `MBU` and `MFU`, as seen in the example output above.    

Happy to adapt the API however useful.